api/src/tool/console/interactive.py

```diff
@@ -0,0 +1,128 @@
+#
+# Interactive console for using database and services.
+#
+# Connects to the database then drops into a Python interactive prompt.
+#
+# Use via the `-i` flag to python, for example:
+#   poetry run python3 -i -m src.tool.console.interactive
+#
+
+import logging
+from typing import Union
+
+import rich
+import rich.panel
+import rich.pretty
+import sqlalchemy
+import src.db
+import src.db.models
+import src.logging
+import src.util
+import tests.src.db.models.factories
+from sqlalchemy import create_engine
+from sqlalchemy.orm import scoped_session, sessionmaker
+from src.adapters.db.clients.postgres_config import get_db_config
+
+INTRO = """
+Simpler Grants Gov Python console
+
+Useful things:
+
+│ db = database session (Not yet working)
+│ f = DB factories module
+│ u = utilities module
+│ r = function to reload REPL
+
+Tips:
+★ Tab-completion is available
+★ History is available (use ↑↓ keys)
+★ Use Ctrl+D to exit
+"""
+
+
+def interactive_console() -> dict:
+    """Set up variables and print a introduction message for the interactive console."""
+
+    db = connect_to_database()
+
+    print(INTRO.format(**locals()))
+
+    variables = dict()
+    variables.update(vars(src.db.models.opportunity_models))
+
+    # This goes after the imports of entire modules, so the console reserved
+    # names (db, fineos, etc) take precedence. This might break some modules
+    # that expect something different under those names.
+    # variables.update(locals())
+
+    # DB Factories
+    factories_module = tests.src.db.models.factories
+
+    if isinstance(db, sqlalchemy.orm.scoped_session):
+        factories_module.db_session = db
+
+    variables["f"] = tests.src.db.models.factories
+
+    # Easy access to utilities
+    variables["u"] = src.util
+    variables["util"] = src.util
+
+    # Easy reloading of modules imported in REPL, for retrying something after a
+    # code change without dropping out of REPL
+    variables["r"] = reload_repl
+    variables["reload"] = reload_repl
+
+    variables["reload_module"] = reload_module
+
+    return variables
+
+
+def connect_to_database():
+    db_config = get_db_config()
+    db_config.hide_sql_parameter_logs = False
+    db: Union[sqlalchemy.orm.scoped_session, Exception]
+    try:
+        db = tests.src.db.init(config=db_config, sync_lookups=True)
+    except Exception as err:
+        db = err
+
+    logging.getLogger("sqlalchemy.engine").setLevel(logging.INFO)  # noqa: B1
+    return db
```

Comment:
The database and logging setup is a bit different from the prior project where this was pulled. A few notes:
* logging doesn't need the `noqa` comments anymore - we simplified the logging and expect to use pythons logging library directly
* On the note of logging, if you want the loggers to work, you'll need to initialize it - see https://github.com/HHS/simpler-grants-gov/blob/main/documentation/api/monitoring-and-observability/logging-configuration.md - like just add this `with` block around the code in the `__name__ == "__main__"` function below
* The database initialization is also quite different + lookup values are synced to the DB in a new way (that I don't think you'd need to worry about for this). Instead we now use a `db_client` which you can get by just initializing a `PostgresDBClient` class with the config you adjusted here. We should return that client from this function.